### PR TITLE
Set out channel axis per layer

### PR DESCRIPTION
**Pull request by eladc-git**
This PR will fix Transformer based pytorch networks.
Fix issue: #250

### CODE REVIEW

File: model_compression_toolkit/core/pytorch/reader/graph_builders.py [hunk 1]
```diff
@@ -182,13 +182,20 @@ def nodes_builder(model: GraphModule,
         else:
             graph_node_type = BaseNode
             kwargs = {}
+
+        output_channel_index = ChannelAxis.NCHW.value
```
Comment (haihabi) [inline model_compression_toolkit/core/pytorch/reader/graph_builders.py:186]: This look like a patch, let's talk about this change please don't merge it until we talk @lior-dikstein, @ofirgo, @eladc-git .

Comment (eladc-git) [reply model_compression_toolkit/core/pytorch/reader/graph_builders.py]: I removed it from graph builder and create a dictionary.

**Comment by haihabi** [comment]
@elad-c please link the PR to the issue

**Comment by elad-c** [comment]
> @elad-c please link the PR to the issue

@eladc-git ?

File: model_compression_toolkit/core/keras/default_framework_info.py [hunk 2]
```diff
@@ -111,4 +123,4 @@
                                    ACTIVATION2MINMAX,
                                    LAYER2MINMAX,
                                    KERNEL_ATTRIBUTES,
-                                   OUTPUT_CHANNEL_INDEX)
```
Comment (ofirgo) [inline model_compression_toolkit/core/keras/default_framework_info.py:114]: Please remove the variable OUTPUT_CHANNEL_INDEX if it is not being used anywhere else

File: model_compression_toolkit/core/pytorch/default_framework_info.py [hunk 3]
```diff
@@ -95,4 +105,4 @@
                                      ACTIVATION2MINMAX,
                                      LAYER2MINMAX,
                                      KERNEL_ATTRIBUTES,
-                                     OUTPUT_CHANNEL_INDEX)
```
Comment (ofirgo) [inline model_compression_toolkit/core/pytorch/default_framework_info.py:98]: Please remove the variable OUTPUT_CHANNEL_INDEX if it is not being used anywhere else

File: model_compression_toolkit/core/keras/default_framework_info.py [hunk 4]
```diff
@@ -52,6 +52,18 @@
                                          Dense: (1, 0),
                                          Conv2DTranspose: (2, 3)}, lambda: (None, None))
 
+
+"""
+Map a layer to its output channel axis. 
+Where axis=-1 is the last axis
+"""
+DEFAULT_OUT_CHANNEL_AXIS_DICT = DefaultDict({Conv2D: -1,
```
Comment (ofirgo) [inline model_compression_toolkit/core/keras/default_framework_info.py:60]: Consider to use constant variable "LAST_AXIS" instead of const value -1

File: model_compression_toolkit/core/pytorch/default_framework_info.py [hunk 5]
```diff
@@ -47,11 +47,21 @@
                                          ConvTranspose2d: (1, 0)},
                                         lambda: (None, None))
 
+"""
+Map a layer to its output channel axis.
+Where axis=-1 is the last axis
+"""
+DEFAULT_OUT_CHANNEL_AXIS_DICT = DefaultDict({Conv2d: 1,
+                                             Linear: -1,
```
Comment (ofirgo) [inline model_compression_toolkit/core/pytorch/default_framework_info.py:55]: Consider to use constant variable "LAST_AXIS" instead of const value -1

**Review by ofirgo — CHANGES_REQUESTED** [review]
Only cosmetic changes to consider

File: model_compression_toolkit/core/keras/default_framework_info.py (hunk 2 above)
Comment (eladc-git) [reply model_compression_toolkit/core/keras/default_framework_info.py:114]: you are looking an old version. It was replaced by the dict variable.

File: model_compression_toolkit/core/pytorch/default_framework_info.py (hunk 3 above)
Comment (eladc-git) [reply model_compression_toolkit/core/pytorch/default_framework_info.py:98]: same

File: model_compression_toolkit/core/common/collectors/statistics_collector.py [hunk 6]
```diff
@@ -56,25 +56,25 @@ class StatsCollector(BaseStatsCollector):
...
+                 out_channel_axis: int,
                  init_min_value: float = None,
                  init_max_value: float = None):
         """
         Instantiate three statistics collectors: histogram, mean and min/max per channel.
         Set initial min/max values if are known.
 
         Args:
-            output_channel_index: Index of output channels.
+            axis: Index of output channels.
```
[the remark this answers is not in the document]
Comment (eladc-git) [reply model_compression_toolkit/core/common/collectors/statistics_collector.py:67]: fixed.

File: model_compression_toolkit/core/common/collectors/mean_collector.py [hunk 7]
```diff
@@ -91,7 +91,7 @@ def update(self,
         """
 
         self.i += 1  # Update the iteration index
-        axis = (len(x.shape) - 1) if self.axis == ChannelAxis.NHWC else self.axis.value  # convert
+        axis = (len(x.shape) - 1) if self.axis == -1 else self.axis
```
[the remark this answers is not in the document]
Comment (eladc-git) [reply model_compression_toolkit/core/common/collectors/mean_collector.py:94]: done

File: model_compression_toolkit/core/keras/default_framework_info.py (hunk 2 above)
Comment (ofirgo) [reply model_compression_toolkit/core/keras/default_framework_info.py:114]: Isn't the actual old variable still exists (on line 118?), I see that it's being assigned there

**Review by ofirgo — APPROVED** [review]
